chainer/links/connection/convolution_2d.py

```diff
@@ -227,8 +238,14 @@ def forward(self, x):
             ~chainer.Variable: Output of the convolution.
 
         """
-        if self.W.array is None:
-            self._initialize_params(x.shape[1])
+        x = chainer.as_variable(x)
+        assert x.layout == self.x_layout
+        # self.W can be a Variable instead of Parameter: #8462
+        # TODO(niboshi): Use Parameter.is_initialized.
+        if self.W.raw_array is None:
```

Comment:
I fixed to avoid assuming that `self.W` is a `Parameter`.
#8462 is no longer required.